### PR TITLE
Incorrect location indicator in settings page due to disabled location permission.

### PR DESCRIPTION
**Pull request by shilpashendre**
Location indicator is set properly in the settings page.

### CODE REVIEW

**Review by Patrick-Erichsen — CHANGES_REQUESTED** [review]
Hey @shilpashendre , is there any relevant Github issue you could link on this? I don't quite understand what the fix is for.

Can you also provide testing instructions to verify that the fix is working correctly?

File: gps/app/views/Settings.js [hunk 1]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+      } else {
+        // when user deny
+        setIsLogging(false);
+        LocationServices.stop();
+        await SetStoreData(PARTICIPATE, false);
+      }
+    });
+  };
+
+  // checking whether location permission is enable or disbale from application setting on load of setting's page.
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]:
```suggestion
  // checking whether location permission is enabled or disabled from application setting on a load of setting's page.
```

File: gps/app/views/Settings.js [hunk 2]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+  const checkIsLocationEnable = () => {
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]:
```suggestion
  const isLocationTrackingEnabled = () => {
```

File: gps/app/views/Settings.js [hunk 3]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+    LocationService.checkStatus()
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: Please use the `async`/`await` pattern here.

File: gps/app/views/Settings.js [hunk 4]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+      .then(async res => {
+        if (res !== undefined) {
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: this line can be removed and the next line can just be 
```js
if (res.isRunning) {
```

File: gps/app/views/Settings.js [hunk 5]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+          if (res.isRunning === true) {
+            await SetStoreData(PARTICIPATE, true);
+            setIsLogging(true);
+          }
+        }
+      })
+      .catch(err => {
+        console.log('TCL: SettingsScreen -> err', err);
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: What is `TCL`?

File: gps/app/views/Settings.js [hunk 6]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
     };
   }, [navigation]);
 
+  //  This method is used to track whether user has "Allow" or "Deny" location permission system Dialog.
+  const systemDialogPermission = async () => {
+    BackgroundGeolocation.on('authorization', async status => {
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: Please use `async`/`await` pattern here.

File: gps/app/views/Settings.js [hunk 7]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+      if (status === BackgroundGeolocation.AUTHORIZED) {
+        // when user allow
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: this comment and the one in the `else` branch can be removed.

File: gps/app/views/Settings.js [hunk 8]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
     };
   }, [navigation]);
 
+  //  This method is used to track whether user has "Allow" or "Deny" location permission system Dialog.
+  const systemDialogPermission = async () => {
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: I'm confused about what this method is doing. Is there a dialog that is being prompted?

File: gps/app/views/Settings.js (hunk 4 above)
Comment (shilpashendre) [reply gps/app/views/Settings.js:78]: Hi, TCL is a Turbo console log extension, use as a shortcut to create console prints.

File: gps/app/views/Settings.js (hunk 5 above)
Comment (shilpashendre) [reply gps/app/views/Settings.js]: Hi, TCL is a Turbo console log extension, use as a shortcut to create console prints. I will remove it.

File: gps/app/views/Settings.js (hunk 6 above)
Comment (Patrick-Erichsen) [reply gps/app/views/Settings.js]: I didn't notice in my first review that this is using the `BackgroundGeolocation` library.

The current file is for the `Settings` screen - we want to try and keep this file limited to logic that is only related to the UI content of this screen.

This logic should be moved to the [existing authorization handler in `LocationServices.js`](https://github.com/Path-Check/covid-safe-paths/blob/develop/gps/app/services/LocationService.js#L159).

The view logic for this method (e.g. `setIsLogging()`) should stay in this file but pull from LocationService.

File: gps/app/views/Settings.js (hunk 8 above)
Comment (Patrick-Erichsen) [reply gps/app/views/Settings.js]: Could you provide a more descriptive name for this method?

File: gps/app/views/Settings.js [hunk 9]
```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
...
+        console.log('TCL: SettingsScreen -> err', err);
+      });
+  };
+
   const locationToggleButtonPressed = async () => {
     try {
       isLogging ? LocationServices.stop() : LocationServices.start();
       await SetStoreData(PARTICIPATE, !isLogging);
       setIsLogging(!isLogging);
+      systemDialogPermission();
```
Comment (Patrick-Erichsen) [inline gps/app/views/Settings.js]: Why do we need this method at all if all it is doing is calling (either true of false values)

```js
setIsLogging(false);
LocationServices.stop();
 await SetStoreData(PARTICIPATE, false);
```

which we have already done in the three lines above?

**Review by Patrick-Erichsen — CHANGES_REQUESTED** [review]

File: app/views/Settings.js [hunk 10]
```diff
@@ -53,11 +50,47 @@ export const SettingsScreen = ({ navigation }) => {
...
+          'Something went wrong in system dialog location permission',
+          err,
+        );
+      });
+  };
+
+  // checking whether location permission is enable or disbale from application setting on load of setting's page.
+  const checkIsLocationEnable = async () => {
+    await LocationServices.checkStatus()
+      .then(async res => {
```
Comment (Patrick-Erichsen) [inline app/views/Settings.js]: .then is not needed with await

File: app/services/LocationService.js [hunk 11]
```diff
@@ -258,6 +258,17 @@ export default class LocationServices {
     });
   }
 
+  // To get the system dialog click event
+  static async getSystemLocationPermission() {
```
Comment (Patrick-Erichsen) [inline app/services/LocationService.js:262]: is this being used anywhere?

**Review by Patrick-Erichsen — CHANGES_REQUESTED** [review]

File: app/services/LocationService.js (hunk 11 above)
Comment (shilpashendre) [reply app/services/LocationService.js:262]: No. Only one time in the settings page.

File: app/views/Settings.js [hunk 12]
```diff
@@ -55,6 +52,41 @@ export const SettingsScreen = ({ navigation }) => {
...
+      })
+      .catch(err => {
+        console.log('Is background Location enable ', err);
+      });
+  };
+
+  //  This method is used to track whether user has "Allow" or "Deny" location permission system Dialog.
+  const systemDialogLocationPermission = async () => {
+    await LocationServices.getSystemLocationPermission()
+      .then(res => {
```
Comment (Patrick-Erichsen) [inline app/views/Settings.js]: This isn't quite right - I think taking a look at this link may be helpful https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

The short answer is that `await` does not require a `.then()`.

File: app/services/LocationService.js (hunk 11 above)
Comment (Patrick-Erichsen) [reply app/services/LocationService.js:262]: I still don't quite understand why this is needed. There is no callback that is passed to the authorization handler, so nothing will ever be done

**Review by Patrick-Erichsen — CHANGES_REQUESTED** [review]

**Comment by Patrick-Erichsen** [comment]
Closing as this is no longer relevant with the work being done in #945

**Pull request closed without merging**